flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -2974,6 +2971,137 @@ public void testSharedStateRegistrationOnRestore() throws Exception {
         }
     }
 
+    @Test
+    public void testSharedStateRegistrationWithoutRebuildSharedStateRegistry() throws Exception {
```

Comment:
This test tests way more than it claims to do. Moreover I don't think it tests the no-rebuilding at all. None of the methods from `CheckpointRecoveryFactory` are used here.

Reply:
I agree with Dawid, it tests the existing functionality but the one that was changed.

I think `CheckpointRecoveryFactory` implementations should be unit-tested. And ideally, their use by schedulers too (maybe using `TestingCheckpointRecoveryFactory`).
WDYT?